src/libraries/System.Private.CoreLib/src/System/Number.NumberToFloatingPointBits.cs

```diff
@@ -764,5 +1551,177 @@ private static bool ShouldRoundUp(bool lsbBit, bool roundBit, bool hasTailBits)
             // two is even (this is "round-to-even").
             return roundBit && (hasTailBits || lsbBit);
         }
+
+        internal struct AdjustedMantissa
+        {
+            internal ulong mantissa;
+            internal int power2; // a negative value indicates an invalid result
+            public static bool operator ==(AdjustedMantissa a, AdjustedMantissa b)
+                 => a.mantissa == b.mantissa && a.power2 == b.power2;
+
+            public static bool operator !=(AdjustedMantissa a, AdjustedMantissa b)
+               => a.mantissa != b.mantissa || a.power2 != b.power2;
+
+            public override bool Equals(object? obj) => base.Equals(obj);
+
+            public override int GetHashCode() => base.GetHashCode();
+        }
```

Comment:
Could you please add an empty line here and in other places between functions/classes?
```
foo1
{
}
foo2
{
}
```